#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("z_p_gain", double_t, 0, "Z P Gain", 100.0, 0.0, 200.0)
gen.add("z_i_gain", double_t, 0, "Z I Gain", 0.0, 0.0, 200.0)
gen.add("z_d_gain", double_t, 0, "Z D Gain", 75.0, 0.0, 200.0)
gen.add("yaw_p_gain", double_t, 0, "Yaw P Gain", 0.0, 0.0, 200.0)
gen.add("yaw_i_gain", double_t, 0, "Yaw I Gain", 0.0, 0.0, 200.0)
gen.add("yaw_d_gain", double_t, 0, "Yaw D Gain", 0.0, 0.0, 200.0)

gen.add("roll_p_gain", double_t, 0, "Roll P Gain", 0.0, 0.0, 200.0)
gen.add("roll_i_gain", double_t, 0, "Roll I Gain", 0.0, 0.0, 200.0)
gen.add("roll_d_gain", double_t, 0, "Roll D Gain", 0.0, 0.0, 200.0)
gen.add("pitch_p_gain", double_t, 0, "Pitch P Gain", 0.0, 0.0, 200.0)
gen.add("pitch_i_gain", double_t, 0, "Pitch I Gain", 0.0, 0.0, 200.0)
gen.add("pitch_d_gain", double_t, 0, "Pitch D Gain", 0.0, 0.0, 200.0)

exit(gen.generate(
    "auv_control", "wrench_controller", "WrenchController"))
